pkg/users/account-create-dialog.js

```diff
@@ -222,57 +301,60 @@
                     return !has_errors(errs);
                 });
     }
 
-    function create(real_name, user_name, password, locked, force_change) {
-        return cockpit.spawn(["/usr/sbin/useradd", "-D"], { superuser: "require" })
-                .catch(() => "")
-                .then(defaults => {
-                    let shell = null;
-                    defaults.split("\n").forEach(item => {
-                        if (item.indexOf("SHELL=") === 0) {
-                            shell = item.split("=")[1] || "";
-                        }
-                    });
-                    const prog = ["/usr/sbin/useradd", "--create-home", "-s", shell || "/bin/bash"];
-                    if (real_name) {
-                        prog.push('-c');
-                        prog.push(real_name);
-                    }
-                    prog.push(user_name);
-                    return cockpit.spawn(prog, { superuser: "require", err: "message" })
-                            .then(() => passwd_change(user_name, password))
-                            .then(() => {
-                                if (locked)
-                                    return cockpit.spawn([
-                                        "/usr/sbin/usermod",
-                                        user_name,
-                                        "--lock"
-                                    ], { superuser: "require", err: "message" });
-                                if (force_change)
-                                    return cockpit.spawn([
-                                        "/usr/bin/passwd",
-                                        "-e",
-                                        user_name
-                                    ], { superuser: "require", err: "message" });
-                            });
+    function create(real_name, user_name, password, locked, force_change, home_dir, base_home_dir) {
+        const prog = ["/usr/sbin/useradd", "--create-home", "-s", state.shell];
+        if (real_name) {
+            prog.push('-c');
+            prog.push(real_name);
+        }
+
+        if (home_dir) {
+            prog.push('-d');
+            prog.push(home_dir);
+        }
+        prog.push(user_name);
+        return cockpit.spawn(prog, { superuser: "require", err: "message" })
+                .then(() => passwd_change(user_name, password))
+                .then(() => {
+                    if (locked)
+                        return cockpit.spawn([
+                            "/usr/sbin/usermod",
+                            user_name,
+                            "--lock"
+                        ], { superuser: "require", err: "message" });
+                    if (force_change)
+                        return cockpit.spawn([
+                            "/usr/bin/passwd",
+                            "-e",
+                            user_name
+                        ], { superuser: "require", err: "message" });
```

Comment:
These 5 added lines are not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18234-20230131-135835-5b43f8e5-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)